feat(#127) 회원탈퇴모달 API 연동 + Textarea.tsx 내 disabled 추가

ella6010 opened the pull request:
## 🔧 관련 이슈

- 이 PR은 다음 이슈와 관련 있습니다: #127

## 🔄 변경 사항

- [x] 기능 추가


## ✔️ 변경 사항 상세 설명

- 추가된 파일:  `src/api/user-withdrawal/api.ts`, `src/api/user-withdrawal/types.ts`
- 주요 구현/수정 내용:
  - 회원탈퇴 api

## 📸 스크린샷 (UI 변경 시 필수)



## 📝 문서화



## 🔍 리뷰어에게 요청 사항 (선택)



## ⚠️ 기타 주의 사항

ella6010 committed:
✨ feat: 회원 탈퇴 모달 API 연동(#127)

회원 탈퇴 모달 API 연동 + Textarea.tsx 내 disabled 추가

ella6010 committed:
💡 chore: import 경로 재설정

import 경로 재설정

jngmnj commented on src/components/myPage/WithdrawalModal.tsx:
```diff
@@ -25,9 +34,30 @@ const WithdrawalModal = ({ onClose, onConfirm }: WithdrawalModalProps) => {
+      onClose()
+    } catch (error) {
+      console.error('탈퇴 실패:', error)
+      alert('탈퇴 요청 중 오류가 발생했습니다. 다시 시도해 주세요.')
```
alert쓰는부분은 toast사용하면좋겟습니다
no-console 린트설정이 있어서 console.error 사용시 위에 
```
// eslint-disable-next-line no-console
console.error(`[ERROR] ${error ?? ''}`)
```
이런식으로 린트무효화 주석 필요합니다. 

error 타입지정도 해주세용 
[TypeScript에서 axios error 핸들링하기](https://velog.io/@quack777/TypeScript%EC%97%90%EC%84%9C-axios-error-%ED%95%B8%EB%93%A4%EB%A7%81%ED%95%98%EA%B8%B0)

ella6010 replied:
수정했어요~!

jngmnj commented on src/components/myPage/WithdrawalModal.tsx:
```diff
@@ -25,9 +34,30 @@ const WithdrawalModal = ({ onClose, onConfirm }: WithdrawalModalProps) => {
+      }
+
+      const res = await UserWithdrawalApi.withdraw(payload)
+      console.log('탈퇴 완료:', res)
```
console.log 삭제부탁드립니닷 빌드에러납니다

ella6010 replied:
수정했어요~!

jngmnj commented on src/components/myPage/WithdrawalModal.tsx:
```diff
@@ -17,6 +18,14 @@ const reasons = [
+  const toggleDropdown = () => setDropdownOpen((prev) => !prev)
+
+  // 이메일은 예시용, 실제로는 props나 전역 상태에서 받아야 합니다.
+  const userEmail = 'user@example.com'
```
지금 api 연동인데 하드코딩돼있어서 수정필요합니닷
const userEmail = useAuthStore().user?.email

ella6010 replied:
수정했어요~!

ella6010 committed:
💡 chore: alert → toast 변경, 하드코딩된 userEmail → useAuthStore().user?.email로 변경

- alert → toast로 변경
- console.error → // eslint-disable-next-line 주석 추가 + 포맷 통일
- 하드코딩된 userEmail → useAuthStore().user?.email로 변경
- error 타입 지정 추가

jngmnj reviewed (approved):
고생하셨습니다!

Aprasaks reviewed (approved):
절대경로 한번 다시 돌려야할듯요 저희 ㅠ
